beginend.el

```diff
@@ -239,6 +239,13 @@ BEGIN-BODY and END-BODY are two `progn' expressions passed to respectively
   (progn
     (prodigy-last)))
 
+(beginend-define-mode magit-status-mode
+  (progn
+    (re-search-forward "^$")
+    (forward-line))
+  (progn
+    (beginend--goto-nonwhitespace)))
```

Comment:
If you have the section `Unstaged changes` as the last one (or `Staged changes`) and the last change set is collapsed, the cursor will jump to different positions on repeated invocations.